### ISSUE TITLE
[Bug] --version and --help are exiting with failure code

### ISSUE THREAD

**Issue by dandavison** [temporalio/cli#495]
```
$ /Users/dan/src/temporalio/cli/temporal-new --version > /dev/null; echo $?
unknown command
1

$ /Users/dan/src/temporalio/cli/temporal-new --help > /dev/null; echo $?
unknown command
1
```

**Comment by lorensr** [comment]
Would also be nice if it didn't print `unknown command`

**Comment by josh-berry** [comment]
I spent some time looking into this, and I don't think it's easily fixable unless the following two Cobra issues are fixed:

- https://github.com/spf13/cobra/issues/1156
- https://github.com/spf13/cobra/issues/706

I there's a hack we can put in to detect if help text is being displayed, and use that to `exit 0`, but it would require _not_ printing help text on an invalid subcommand, and I haven't yet figured out how to control that behavior. (I've been poking at this in between meetings and such so haven't really given it a lot of focused attention though.)

tl;dr Still looking...
